.github/workflows/build-nginx-proxy-image.yml

```diff
@@ -0,0 +1,174 @@
+name: Build Nginx Proxy Docker Image
+
+env:
+  IMAGE_NAME: nginx-proxy
+  AWS_REGION: eu-west-1
+
+on:
+  pull_request:
+  push:
```

Comment:
This workflow should be split into 2 to match the pattern we have in the other repos which is:

a run-tests.yml file for unit tests (the name doesn't really matter but it's run-tests nearly everywhere) which has

```
on:
  pull_request:
  workflow_call:
```

So it doesn't run on merge to master, only on PRs (and reusable workflow calls)

Then a second workflow called `post-merge.yml` (whose name is Post Merge) which is:

```
on:
  push:
    branches:
      - master
```

In the post-merge you can call the run-tests one (see frontend for an example: https://github.com/alphagov/pay-frontend/blob/master/.github/workflows/post-merge.yml#L13)

and then you can use the reusable workflow we have for tagging the alpha release (which you already are 😄 in this file but commented out 👍 )

If you want to make cat's life easier :D You can also add the concurrency to the post-merge like this: https://github.com/alphagov/pay-stubs/blob/master/.github/workflows/release-master.yml#L11 (but call it nginx-proxy-post-merge but technically that's part of her story :) I'm sure she would appreciate it though 😄 

A couple of big advantages with this pattern:

You have 2 different Workflows in the UI and it's trivial to separate the Post Merge "releases" workflow runs from the unit tests on PRs.
Using the reusable workflows makes our maintenance lives a lot easier 😄 

I'm happy to pair on this if you want, we've moved pretty far in the short time you've been out, but if you don't feel up to it at the moment that's cool too 👍